addon/validators/base.js

```diff
@@ -118,32 +111,23 @@ const Base = Ember.Object.extend({
     this.value = builtOptions.value || this.value;
     delete builtOptions.value;
 
-    return builtOptions;
-  },
+    const OptionsClass = Ember.Object.extend(builtOptions, {
+      model: computed(() => get(this, 'model')).readOnly(),
+      attribute: computed(() => get(this, 'attribute')).readOnly(),
 
-  /**
-   * Creates a new object and calls any option property that is a function with the validator context.
-   * This method is called right before `validate` and the returned object gets passed into the validate method as its options
-   * @method processOptions
-   * @return {Object}
-   */
-  processOptions() {
-    const options = assign({}, get(this, 'options') || {});
-    const model = get(this, 'model');
-    const attribute = get(this, 'attribute');
+      copy(deep) {
+        if(deep) {
+          return OptionsClass.create();
+        }
 
-    Object.keys(options).forEach(key => {
-      const option = options[key];
-
-      if (typeof option === 'function' && key !== 'message') {
-        options[key] = option.call(this, model, attribute);
+        return Ember.Object.create(Object.keys(builtOptions).reduce((obj, o) => {
+          obj[o] = get(this, o);
+          return obj;
+        }, {}));
       }
     });
 
-    // Cache the options so it can be accessed without triggering this method again
-    set(this, '_cachedOptions', assign({}, options));
-
-    return options;
+    return OptionsClass.create();
```

Comment:
You can go with `Ember.Freezable`, but it's not quite the same as `Object.freeze`.

Or just leave it as is and live by the `readOnly` CPs, which, IMO, is good enough.